delta/sdk-testkit/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/testkit/ResourcesDummy.scala

```diff
@@ -0,0 +1,178 @@
+
+  override def create(
+      project: Project,
+      schema: ResourceRef,
+      source: Json
+  )(implicit caller: Subject): IO[ResourceRejection, DataResource] =
+    sourceAsJsonLD(project, source).flatMap { case (id, compacted, expanded) =>
+      eval(CreateResource(id, project.ref, schema, source, compacted, expanded, caller))
+    }
+
+  override def create(
+      id: Iri,
+      project: ProjectRef,
+      schema: ResourceRef,
+      source: Json
+  )(implicit caller: Subject): IO[ResourceRejection, DataResource] =
+    sourceAsJsonLD(id, source).flatMap { case (compacted, expanded) =>
+      eval(CreateResource(id, project, schema, source, compacted, expanded, caller))
+    }
+
+  override def update(
+      id: Iri,
+      project: ProjectRef,
+      schemaOpt: Option[ResourceRef],
+      rev: Long,
+      source: Json
+  )(implicit caller: Subject): IO[ResourceRejection, DataResource] =
+    sourceAsJsonLD(id, source).flatMap { case (compacted, expanded) =>
+      eval(UpdateResource(id, project, schemaOpt, source, compacted, expanded, rev, caller))
+    }
+
+  override def tag(
+      id: Iri,
+      project: ProjectRef,
+      schemaOpt: Option[ResourceRef],
+      tag: Label,
+      tagRev: Long,
+      rev: Long
+  )(implicit caller: Subject): IO[ResourceRejection, DataResource] =
+    eval(TagResource(id, project, schemaOpt, tagRev, tag, rev, caller))
+
+  override def deprecate(
+      id: Iri,
+      project: ProjectRef,
+      schemaOpt: Option[ResourceRef],
+      rev: Long
+  )(implicit caller: Subject): IO[ResourceRejection, DataResource] =
+    eval(DeprecateResource(id, project, schemaOpt, rev, caller))
+
+  override def fetch(id: Iri, project: ProjectRef, schemaOpt: Option[ResourceRef]): UIO[Option[DataResource]] =
+    cache.fetch((project, id)).map(validateSameSchema(_, schemaOpt))
+
+  override def fetchAt(
+      id: Iri,
+      project: ProjectRef,
+      schemaOpt: Option[ResourceRef],
+      rev: Long
+  ): IO[RevisionNotFound, Option[DataResource]] =
+    journal
+      .stateAt((project, id), rev, Initial, Resources.next, RevisionNotFound.apply)
+      .map(_.flatMap(_.toResource))
+      .map(validateSameSchema(_, schemaOpt))
+
+  private def eval(cmd: ResourceCommand): IO[ResourceRejection, DataResource] =
+    semaphore.withPermit {
+      for {
+        state <- journal.currentState((cmd.project, cmd.id), Initial, Resources.next).map(_.getOrElse(Initial))
+        event <- Resources.evaluate(fetchSchema)(state, cmd)
+        _     <- journal.add(event)
+        res   <- IO.fromEither(Resources.next(state, event).toResource.toRight(UnexpectedInitialState(cmd.id)))
+        _     <- cache.setToCache(res)
+      } yield res
+    }
+
+  override def events(offset: Offset): fs2.Stream[Task, Envelope[ResourceEvent]] =
+    journal.events(offset)
+
+  override def currentEvents(offset: Offset): fs2.Stream[Task, Envelope[ResourceEvent]] =
+    journal.currentEvents(offset)
+
+  private def sourceAsJsonLD(
```

Comment:
We should have the same thing in the real implementation no ?

Reply:
Yes. I will duplicate the code.